packages/time-picker/test/value-commit.test.js

```diff
@@ -0,0 +1,391 @@
+import { fixtureSync, nextRender, outsideClick } from '@vaadin/testing-helpers';
+import { sendKeys } from '@web/test-runner-commands';
+import sinon from 'sinon';
+import '../src/vaadin-time-picker.js';
+import { getAllItems } from './helpers.js';
+
+describe('value commit', () => {
+  let timePicker, valueChangedSpy, validateSpy, changeSpy;
+
+  function expectNoValueCommit() {
+    expect(valueChangedSpy).to.be.not.called;
+    expect(validateSpy).to.be.not.called;
+    expect(changeSpy).to.be.not.called;
+  }
+
+  function expectValueCommit(value) {
+    expect(valueChangedSpy).to.be.calledOnce;
+    expect(validateSpy).to.be.calledOnce;
+    expect(validateSpy.firstCall).to.be.calledAfter(valueChangedSpy.firstCall);
+    expect(changeSpy).to.be.calledOnce;
+    expect(changeSpy.firstCall).to.be.calledAfter(validateSpy.firstCall);
+    expect(timePicker.value).to.equal(value);
+  }
+
+  function expectValidationOnly() {
+    expect(valueChangedSpy).to.be.not.called;
+    expect(validateSpy).to.be.calledOnce;
+    expect(changeSpy).to.be.not.called;
+  }
+
+  beforeEach(async () => {
+    timePicker = fixtureSync(`<vaadin-time-picker></vaadin-time-picker>`);
+    await nextRender();
+    validateSpy = sinon.spy(timePicker, 'validate').named('validateSpy');
+
+    valueChangedSpy = sinon.spy().named('valueChangedSpy');
+    timePicker.addEventListener('value-changed', valueChangedSpy);
+
+    changeSpy = sinon.spy().named('changeSpy');
+    timePicker.addEventListener('change', changeSpy);
+
+    timePicker.focus();
+  });
+
+  describe('default', () => {
+    it('should not commit but validate on blur', () => {
+      timePicker.blur();
+      expectValidationOnly();
+    });
+
+    it('should not commit but validate on Enter', async () => {
+      await sendKeys({ press: 'Enter' });
+      expectValidationOnly();
+    });
+
+    it('should not commit on Escape', async () => {
+      await sendKeys({ press: 'Escape' });
+      expectNoValueCommit();
+    });
+
+    it('should not commit but validate on close with outside click', () => {
+      timePicker.click();
+      outsideClick();
+      expectValidationOnly();
+    });
+
+    it('should not commit but validate on close with Escape', async () => {
+      timePicker.click();
+      await sendKeys({ press: 'Escape' });
+      expectValidationOnly();
+    });
+
+    it('should not commit on ArrowDown', async () => {
+      // Open the dropdown
+      await sendKeys({ press: 'ArrowDown' });
+      // Focus an item
+      await sendKeys({ press: 'ArrowDown' });
+      expectNoValueCommit();
+    });
+
+    it('should not commit on ArrowUp', async () => {
+      // Open the dropdown
+      await sendKeys({ press: 'ArrowUp' });
+      // Focus an item
+      await sendKeys({ press: 'ArrowUp' });
+      expectNoValueCommit();
+    });
+  });
+
+  describe('entering parsable input', () => {
+    beforeEach(async () => {
+      await sendKeys({ type: '12:00' });
```

Comment:
Nit: could be changed to `12`, then there will be no need in second <kbd>Escape</kbd> on line 116.

Reply:
Good point, but I would rather add a comment to explain why we need two <kbd>Escape</kbd> to keep everything as clear as possible.

Reply:
Both options sounds good to me 👌

Reply:
Added a comment.